src/sparseml/exporters/transforms/kv_cache/position_embeddings_adjustment_codegen.py

```diff
@@ -0,0 +1,450 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from copy import deepcopy
+from typing import List, Tuple
+
+import numpy
+import onnx
+from onnx import ModelProto, NodeProto, TensorProto
+
+from sparseml.exporters.transforms.onnx_transform import OnnxTransform
+from sparseml.onnx.utils import ONNXGraph
+from sparseml.onnx.utils.helpers import get_node_output_nodes, get_nodes_by_output_id
+
+
+__all__ = ["PositionEmbeddingsAdjustmentCodeGen"]
+
+
+# regex to find the node that computes position embeddings
+_POSITION_EMBEDDING_PATTERN = r"/transformer/h.\d+/attn/Range_1"
+# regex to find the node that computes rotary position embeddings
+_ROTARY_POSITION_EMBEDDING_PATTERN = r"/transformer/h.\d+/attn/Slice_(10|9|5|4)"
```

Comment:
we should avoid using regex for matching and instead rely on node type patterns where possible - especially important as exported names are subject to change both between torch/onnx versions as well as slight changes to model version/size